src/System.Management.Automation/FormatAndOutput/common/BaseFormattingCommand.cs

```diff
@@ -765,10 +773,27 @@ internal void GetCommandLineProperties(FormattingCommandLineParameters parameter
                 parameters.mshParameterList = processor.ProcessParameters(Property, invocationContext);
             }
 
+            if (ExcludeProperty != null)
+            {
+                parameters.excludePropertyFilter = new PSPropertyExpressionFilter(ExcludeProperty);
+                // ExcludeProperty implies -Property * for better UX
+                if (Property == null || Property.Length == 0)
+                {
+                    CommandParameterDefinition def = isTable
+                        ? new FormatTableParameterDefinition()
+                        : new FormatListParameterDefinition();
+                    ParameterProcessor processor = new ParameterProcessor(def);
+                    TerminatingErrorContext invocationContext = new TerminatingErrorContext(this);
+
+                    parameters.mshParameterList = processor.ProcessParameters(new object[] { "*" }, invocationContext);
```

Comment:
Not for the PR.
We have huge places in our code base where we allocate such temp arrays. We could review this and use modern C# feature `params ReadOnlySpan<YYY>` to pass arguments in methods without allocation.